packages/cli/src/controller/scaffoldgen-controller.ts

```diff
@@ -0,0 +1,232 @@
+import {loadFromJsonOrYaml} from '@subql/common';
+import ejs from 'ejs';
+import {Interface} from 'ethers/lib/utils';
+import * as inquirer from 'inquirer';
+import {upperFirst} from 'lodash';
+import {parseContractPath} from 'typechain';
+import {parseDocument} from 'yaml';
+import {SelectedMethod, UserInput} from '../commands/codegen/generate';
+
+const SCAFFOLD_HANDLER_TEMPLATE_PATH = path.resolve(__dirname, '../template/scaffold-handlers.ts.ejs');
+const ROOT_MAPPING_DIR = 'src/mappings';
+const DEFAULT_HANDLER_BUILD_PATH = './dist/index.js';
+
+interface TopicsFilter {
+  topics: string[];
+}
+interface FunctionFilter {
+  function: string;
+}
+type Filter = TopicsFilter | FunctionFilter;
+
+interface HandlerType {
+  handler: string;
+  kind: string;
+  filter: Filter;
+}
+
+interface DatasourceProp {
+  kind: string;
+  startBlock: number;
+  options: {
+    abi: string;
+    address?: string;
+  };
+  assets: {
+    [key: string]: {
+      file: string;
+    };
+  };
+  mapping: {
+    file: string;
+    handlers: HandlerType[];
+  };
+}
+
+export async function promptSelectables(
+  input: string,
+  availableMethods: string[],
+  memArr: string[]
+): Promise<string[]> {
+  if (input === '*') {
+    return availableMethods;
+  }
+
+  if (input) {
+    try {
+      const chosenFn = await inquirer.prompt({
+        name: 'functions',
+        message: 'Select Functions',
+        type: 'checkbox',
+        choices: availableMethods,
+      });
+      memArr.push(...chosenFn.functions);
+    } catch (e) {
+      throw new Error(e);
+    }
+  }
+  return memArr;
+}
+
+export async function renderTemplate(templatePath: string, outputPath: string, templateData: ejs.Data): Promise<void> {
+  const data = await ejs.renderFile(templatePath, templateData);
+  await fs.promises.writeFile(outputPath, data);
+}
+
+export function getAbiInterface(projectPath: string, abiPath: string): Interface {
+  const abi = loadFromJsonOrYaml(path.join(projectPath, abiPath)) as any;
+  return new Interface(abi);
+}
+
+export function getAvailableEvents(abiInterface: Interface): {[p: string]: EventFragment} {
+  return abiInterface.events;
+}
+
+export function filterObjectsByStateMutability(obj: {[p: string]: FunctionFragment}): {[p: string]: FunctionFragment} {
+  const filteredObject: {[p: string]: FunctionFragment} = {};
+  for (const key in obj) {
+    if (obj[key].stateMutability !== 'view') {
+      filteredObject[key] = obj[key];
+    }
+  }
+  return filteredObject;
+}
+
+export function getAvailableFunctions(abiInterface: Interface): {[p: string]: FunctionFragment} {
+  return filterObjectsByStateMutability(abiInterface.functions);
+}
+
+function constructDatasources(userInput: UserInput): DatasourceProp {
+  const abiName = parseContractPath(userInput.abiPath).name;
+  const formattedHandlers: HandlerType[] = [];
+
+  userInput.functions.map((fn) => {
+    const handler: HandlerType = {
+      handler: `handle${upperFirst(fn.name)}_${abiName}Tx`,
```

Comment:
Lets be consistent with naming 
```suggestion
      handler: `handle${upperFirst(fn.name)}${abiName}Tx`,
```